new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1148 @@
+                .clusterName(clusterName)
+                .replicas(replicaCount)
+                .serverStartState("RUNNING")));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, domainNamespace);
+
+    // verify the admin server service created
+    checkServiceExists(adminServerPodName, domainNamespace);
+
+    // verify admin server pod is ready
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+    // verify managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking managed server service {0} is created in namespace {1}",
+          managedServerPodNamePrefix + i, domainNamespace);
+      checkServiceExists(managedServerPodNamePrefix + i, domainNamespace);
+    }
+
+    // verify managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Waiting for managed server pod {0} to be ready in namespace {1}",
+          managedServerPodNamePrefix + i, domainNamespace);
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  //deploy application clusterview.war to domain
+  private void deployApplication(String targets) {
+    logger.info("Getting node port for T3 channel");
+    int t3channelNodePort = assertDoesNotThrow(()
+        -> getServiceNodePort(domainNamespace, adminServerPodName + "-external", "t3channel"),
+        "Getting admin server t3channel node port failed");
+    assertNotEquals(-1, t3ChannelPort, "admin server t3channelport is not valid");
+
+    //deploy application
+    logger.info("Deploying webapp {0} to domain", clusterViewAppPath);
+    deployUsingWlst(K8S_NODEPORT_HOST, Integer.toString(t3channelNodePort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT, targets, clusterViewAppPath,
+        domainNamespace);
+  }
+
+  //restart pods by manipulating the serverStartPolicy to NEVER and IF_NEEDED
+  private void restartDomain() {
+    logger.info("Restarting domain {0}", domainNamespace);
+    shutdownDomain(domainUid, domainNamespace);
+
+    logger.info("Checking for admin server pod shutdown");
+    checkPodDoesNotExist(adminServerPodName, domainUid, domainNamespace);
+    logger.info("Checking managed server pods were shutdown");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodDoesNotExist(managedServerPodNamePrefix + i, domainUid, domainNamespace);
+    }
+
+    startDomain(domainUid, domainNamespace);
+    //make sure that the introspector runs on a cold start
+    verifyIntrospectorRuns();
+    logger.info("Checking for admin server pod readiness");
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    logger.info("Checking for managed servers pod readiness");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  //create a JDBC datasource targeted to cluster.
```

Comment:
A good candidate for common Test Utility. May be we can comeback to refactor

Reply:
Sure, later.